apps/src/lib/node/ledger/shell/queries.rs

```diff
@@ -117,25 +118,51 @@ where
                 }
             }
             Ok((None, _gas)) => {
-                let proof_ops = if is_proven {
-                    match self.storage.get_non_existence_proof(key) {
-                        Ok(proof) => Some(proof.into()),
-                        Err(err) => {
-                            return response::Query {
-                                code: 2,
-                                info: format!("Storage error: {}", err),
-                                ..Default::default()
-                            };
+                if let Some(asset_type) = is_masp_conversion(key) {
```

Comment:
Because of the extra processing, I think for this it may be a bit better to add a new `rpc::Path` so that this handler only returns directly what's written into the storage